### PR TITLE
[1.3.7-prepare][Bug][Master] Failed to obtain JDBC Connection; nested exception is com.alibaba.druid.pool.DataSourceClosedException: dataSource already closed

### PR DESCRIPTION
**Pull request by ruanwenjun**
issue: #5491, #5578 
pr: #5497

### CODE REVIEW

File: dolphinscheduler-service/src/test/java/org/apache/dolphinscheduler/service/zk/ZookeeperCachedOperatorTest.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        };
+        AbstractListener abstractListener2 = new AbstractListener(2) {
+            @Override
+            protected void dataChanged(CuratorFramework client, TreeCacheEvent event, String path) {
+                // ignore
+            }
+        };
+        zookeeperCachedOperator.registerListener(abstractListener2);
+        zookeeperCachedOperator.registerListener(abstractListener1);
+        Assert.assertTrue(true);
```
Comment (Technoboy-) [inline dolphinscheduler-service/src/test/java/org/apache/dolphinscheduler/service/zk/ZookeeperCachedOperatorTest.java:46]: I think the test has tested nothing.

**Review by Technoboy- — CHANGES_REQUESTED** [review]

**Review by CalvinKirs — APPROVED** [review]
LGTM

File: dolphinscheduler-service/src/test/java/org/apache/dolphinscheduler/service/zk/ZookeeperCachedOperatorTest.java (hunk 1 above)
Comment (CalvinKirs) [reply dolphinscheduler-service/src/test/java/org/apache/dolphinscheduler/service/zk/ZookeeperCachedOperatorTest.java:46]: Sorry, I did not see your comment. Yes, this assertion has no meaning.

Comment (ruanwenjun) [reply dolphinscheduler-service/src/test/java/org/apache/dolphinscheduler/service/zk/ZookeeperCachedOperatorTest.java:46]: Yes, you are right, this test just for debug 👍

**Pull request merged**